doc/compiler/memory/MemoryManager.md

```diff
@@ -15,90 +15,378 @@ by the Compiler; there are too many participants for informal trust,
 book-keeping is expensive, and optimizing for divergent component 
 use-cases is difficult to impossible.
 
-Therefore, the compiler manages memory its own memory.
+Therefore, the compiler manages its own memory.
 
 ## Compiler Memory Manager Hierarchy
 
+The compiler memory management implementation is divided into two
+distinct levels: the low level allocators that allocate (and return)
+typically "large" TR::MemorySegments, and the high level allocators
+that provide allocation support for STL containers as well as bulk
+free and region based allocation.
+
+All the classes described in this document reside within the
+`compiler/env` directory of the project. Many of the classes
+are implemented to be *extensible* meaning that downstream
+projects can either reuse the OMR implementations of these classes
+(e.g. the `jitbuilder` project reuses the OMR implementations
+completely), or can extend the OMR project's implementations
+(don't have a current example of this, but not hard to imagine),
+or can provide their own to replace the OMR implementations
+(Eclipse OpenJ9 examplifies this approach).
+
+The remainder of this document describes the OMR implemtation of the
+two levels of memory allocator management in the compiler.
+
 ### Low Level Allocators
-Low Level Allocators cannot be used by STL containers. They are 
-generally used by High Level Allocators as the underlying 
-provider of memory.
+Low level allocators interface directly with whatever memory provider
+is available outside of the JIT. In a languqage runtime environment,
+that may mean allocating memory from services that track memory allocated
+by the virtual machine. Or the memory provider can simply be "malloc".
+At its lowest level, the OMR memory allocator currently employ the
+system equivalent of malloc/mmap.
+
+There are three fundamental pieces to the lowest memory allocation
+layer: `TR::RawAllocator`, `TR::BackingMemoryAllocator`, and
+`TR::SegmentAllocator`. Each of these classes builds upon the
+earlier classes: one constructs a `TR::BackingMemoryAllocator`
```

Comment:
Pedantic: it isn't immediately obvious that "one" here is meant to be the proverbial "one"; because of the previous sentence it could read as "one of the classes constructs a....".

Reply:
agree, will fix

Reply:
that is definitely an awkward sentence that I seem to have missed in my own review....thanks!